freedomofpress/securedrop
# Fix locally built deb packages under staging environment

## Status

Ready for review

## Description of Changes

Updates the staging environment configuration, particularly the package build logic, so that the application code runs under Apache, as intended. Since merging #1464, we now have a tighter loop on testing app code changes under the Apache setup. Removed the strategy where AppArmor profiles run under "complain" (rather than "enforce"), since most of the problems fixed in this PR slipped in due to the complain strategy.

Resolves #1646. 

Changes proposed in this pull request:

* Stop setting "complain" on AppArmor profiles under staging.
* Remove new `util.py` app code file, which didn't have an AppArmor entry.
* Add AppArmor entries for Flask-Asset-generated CSS and JS files, and include those empty directories in the package build logic.
* Don't assume the `CUSTOM_NOTIFICATION` value will be defined in `config.py`; it will in newly installed instances, but not on those that upgraded from a prior version.

## Testing

How should the reviewer test this PR?

1. Create staging environment from scratch and provision it fully.
2. Ensure that the app tests and config tests pass on all staging hosts.

## Deployment

Any special considerations for deployment? Consider both:

These changes are intended to restore the usefulness of the staging environment, so that locally built deb packages can be used to evaluate how code changes affect a prod-like serving environment, including Apache vhosts, Tor Onion Services, and AppArmor profiles. These changes are necessary to review e.g. #1642 and #1643.

## Checklist

### If you made changes to the app code:


### If you made changes to the system configuration:

- [x] Testinfra tests pass on the mon-staging VM
- [x] Testinfra tests pass on the build VM
